modules/axtask/src/api.rs

```diff
@@ -155,6 +155,7 @@ pub fn set_current_affinity(cpumask: AxCpuMask) -> bool {
         current().set_cpumask(cpumask);
         // After setting the affinity, we need to check if current cpu matches
         // the affinity. If not, we need to migrate the task to the correct CPU.
+        #[cfg(feature = "smp")]
         if !cpumask.get(axhal::cpu::this_cpu_id()) {
```

Comment:
what should we do if "smp" feature is not enabled???

Reply:
That's ok, set `cpumask` only and do not migrate.

Reply:
ok, and this PR is ready for review